core/src/snapshot_utils.rs

```diff
@@ -112,58 +160,64 @@ pub fn remove_snapshot<P: AsRef<Path>>(slot: u64, snapshot_path: P) -> Result<()
     Ok(())
 }
 
-pub fn load_snapshots<P: AsRef<Path>>(
-    names: &[u64],
+pub fn load_snapshots(
+    snapshot_paths: &[SlotSnapshotPaths],
     bank0: &mut Bank,
     bank_maps: &mut Vec<(u64, u64, Bank)>,
     status_cache_rc: &StatusCacheRc,
-    snapshot_path: P,
 ) -> Option<u64> {
     let mut bank_root: Option<u64> = None;
 
-    for (i, bank_slot) in names.iter().rev().enumerate() {
-        let snapshot_file_name = get_snapshot_file_name(*bank_slot);
-        let snapshot_dir = get_bank_snapshot_dir(&snapshot_path, *bank_slot);
-        let snapshot_file_path = snapshot_dir.join(snapshot_file_name.clone());
-        trace!("Load from {:?}", snapshot_file_path);
-        let file = File::open(snapshot_file_path);
-        if file.is_err() {
-            warn!("Snapshot file open failed for {}", bank_slot);
+    for (i, slot_paths) in snapshot_paths.iter().rev().enumerate() {
+        let snapshot_file = File::open(&slot_paths.snapshot_file_path);
+        if snapshot_file.is_err() {
+            warn!("Snapshot file open failed for {:?}", slot_paths);
             continue;
         }
-        let file = file.unwrap();
-        let mut stream = BufReader::new(file);
+        let snapshot_file = snapshot_file.unwrap();
+        let mut stream = BufReader::new(snapshot_file);
         let bank: Result<Bank> =
             deserialize_from(&mut stream).map_err(|_| get_io_error("deserialize bank error"));
         let slot: Result<u64> = deserialize_from(&mut stream)
             .map_err(|_| get_io_error("deserialize bank parent error"));
         let parent_slot = if slot.is_ok() { slot.unwrap() } else { 0 };
         let root: Result<u64> =
             deserialize_from(&mut stream).map_err(|_| get_io_error("deserialize root error"));
-        let status_cache: Result<StatusCacheRc> = deserialize_from(&mut stream)
-            .map_err(|_| get_io_error("deserialize bank status cache error"));
+
         if bank_root.is_none() && bank0.rc.update_from_stream(&mut stream).is_ok() {
             bank_root = Some(root.unwrap());
         }
+
+        let status_cache_file = File::open(&slot_paths.snapshot_status_cache_path);
+        if status_cache_file.is_err() {
+            warn!(
+                "Snapshot status cache file open failed for {:?}",
+                slot_paths
+            );
+            continue;
```

Comment:
@carllin I probably shouldn't `continue` here right? do we want to add this bank into the `bank_maps` even if its status cache failed ?

Reply:
hmmm yeah I think if any of the status caches fail here we should exit with an Error. In my PR, this function is renamed to `bank_from_snapshots` and returns a Result.

Reply:
gotcha. I'll add it to my rebase todo list. waiting on #5417 now

Reply:
When we pass in the `snapshot_paths` here, we get it from `snapshot_utils::get_snapshot_paths`. I think we just need to make sure that this directory doesn't contain gibberish that will make us bail here. That should be true in my PR because I create a new snapshots directory everytime we untar the snapshot archive.